pkg/controller/pravegacluster/upgrade_test.go

```diff
@@ -275,6 +279,359 @@ var _ = Describe("Pravega Cluster Version Sync", func() {
 				})
 			})
 		})
+		Context("syncClusterVersion when cluster in upgrading state", func() {
+			var (
+				err          error
+				foundPravega *v1beta1.PravegaCluster
+				client       client.Client
+			)
+			BeforeEach(func() {
+				client = fake.NewFakeClient(p)
+				r = &ReconcilePravegaCluster{client: client, scheme: s}
+				_, err = r.Reconcile(req)
+				foundPravega = &v1beta1.PravegaCluster{}
+				_ = client.Get(context.TODO(), req.NamespacedName, foundPravega)
+				foundPravega.Status.SetUpgradingConditionTrue("UpgradeController", "0")
+				r.client.Update(context.TODO(), foundPravega)
+				err = r.syncClusterVersion(foundPravega)
+			})
+			It("Error should be nil when the target version is Empty", func() {
+				Ω(err).Should(BeNil())
+			})
+			It("Error should be nil when the target version is equal to current version", func() {
```

Comment:
we are only checking err is not nill here maybe we can change the assertion here